### PR TITLE
Add configurable death/retry count

### PR DESCRIPTION
**Pull request by anthony-dinino-aligent**
**Description of the proposed changes**
Add configurable admin retry count/limit, replacing the hard-coded limit in `RetryHandler.php` 
**Screenshots (if applicable)**
image

**Time tracking**
Please use Toggl time code **eg: ALG-119: magento-async-events Code Review**

### CODE REVIEW

File: etc/adminhtml/system.xml [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+<config xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+        xsi:noNamespaceSchemaLocation="urn:magento:module:Magento_Config:etc/system_file.xsd">
+    <system>
+        <section id="system" translate="label" type="text" sortOrder="900" showInDefault="1">
+            <tab>advanced</tab>
+            <resource>Aligent_AsyncEvents::manage</resource>
+            <group id="async_events" translate="label" type="text" sortOrder="1000" showInDefault="1">
+                <label>Async Events</label>
+                <field id="retry" translate="label" type="text" sortOrder="10" showInDefault="1">
+                    <label>Retry Count</label>
```
Comment (gowrizrh) [inline etc/adminhtml/system.xml]: Just realised that we'll need to change this to something like `Death Count` or `Maximum Deaths`

Comment (anthony-dinino-aligent) [reply etc/adminhtml/system.xml:11]: I will change it to Death Count

Comment (anthony-dinino-aligent) [reply etc/adminhtml/system.xml:11]: I will change it to `Death Count` :+1:

Comment (anthony-dinino-aligent) [reply etc/adminhtml/system.xml]: I changed it to max_deaths

**Comment by anthony-dinino-aligent** [comment]
@gowrizrh This hasn't been tested that it actually uses the configurable value yet.

**Review by gowrizrh — APPROVED** [review]

**Pull request merged**